pkg/cptvframe/frame.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2018 The Cacophony Project. All rights reserved.
+// Use of this source code is governed by the Apache License Version 2.0;
+// see the LICENSE file for further details.
+
+package cptvframe
+
+import (
+	"time"
+)
+
+type Telemetry struct {
+	TimeOn       time.Duration
+	FFCState     string
+	FrameCount   int
+	FrameMean    uint16
+	TempC        float64
+	LastFFCTempC float64
+	LastFFCTime  time.Duration
+}
+
+// Frame represents the thermal readings for a single frame.
+type Frame struct {
+	Pix    [][]uint16
+	Status Telemetry
+}
+
+// Interface that all thermal camera implementations should implement
+type CameraResolution interface {
```

Comment:
I suggest that you call this `CameraDefinition`, `CameraSpec` or `CameraDef`.

It's likely that we might also want to include the brand, model name, pixel bit width and frame rate here for inclusion in the CPTV header.

Reply:
Given that this only defines static data with no behaviour it doesn't necessarily have to be an interface (could just be a struct). Your call though - I don't mind too much.

Reply:
Is there any advantages of using a stuct?

Reply:
Instantiating a struct is a easier and more concise than defining a new type with lots of methods, but otherwise no.